pages/index.js

```diff
@@ -1,5 +1,26 @@
 import PlantList from "@/components/PlantList";
+import { plants } from "@/assets/plants";
+import Footer from "@/components/Footer";
+import useLocalStorageState from "use-local-storage-state";
+const initialPlants = plants;
 
 export default function Homepage() {
-  return <PlantList />;
+  const [plants, setPlants] = useLocalStorageState("favourites", {
+    defaultValue: initialPlants,
+  });
+
+  function toggleFavourite(id) {
+    setPlants((prevPlants) =>
+      prevPlants.map((plant) =>
+        plant.id === id ? { ...plant, isFavourite: !plant.isFavourite } : plant
+      )
+    );
+  }
+
+  return (
+    <div>
+      <PlantList plants={plants} toggleFavourite={toggleFavourite} />
+      <Footer />
+    </div>
```

Comment:
- As mentioned above.  Consider using a `<main>` element instead of `<div>` to wrap the primary content of the page.

- Also, if you plan to reuse the layout with a header or other components, consider creating a Layout component to wrap content and ensure consistent structure across pages.
```suggestion
    <main>
      <PlantList plants={plants} toggleFavourite={toggleFavourite} />
      <Footer />
    </main>
```